source/apiVolontaria/volunteer/serializers.py

```diff
@@ -414,11 +418,38 @@ class ParticipationBasicSerializer(serializers.ModelSerializer):
 
     # Explicitly declare the BooleanField to make it "required"
     standby = serializers.BooleanField()
+    invit_icalendar = serializers.BooleanField(
+        default=False,
+        write_only=True,
+    )
     user = UserBasicSerializer(
         read_only=True,
         default=serializers.CurrentUserDefault(),
     )
 
+    def create(self, validated_data):
+        invitation_ics = validated_data.pop('invit_icalendar')
+        participation = models.Participation.objects.create(**validated_data)
+
+        if invitation_ics:
+            if (os.path.exists(settings.MEDIA_ROOT) == False):
```

Comment:
You need to check if the `settings.CONSTANT['EMAIL_SERVICE']` is `True` or send an error because you can't send the ICS for the moment.

Actually i have an error 500 from the API if my email service is not configured